src/core/qgsvectorlayer.cpp

```diff
@@ -3134,30 +3134,26 @@ QString QgsVectorLayer::displayExpression() const
   {
     QString idxName;
 
-    const auto constMFields = mFields;
-    for ( const QgsField &field : constMFields )
+    // Check the fields and keep the first one that matches.
+    // We assume that the user has organized the data with the
+    // more "interesting" field names first. As such, name should
+    // be selected before oldname, othername, etc.
+    // This candidates list is a prioritized list of candidates ranked by "interestingness"!
+    static QStringList sCandidates{ QStringLiteral( "name" ), QStringLiteral( "descript" ), QStringLiteral( "street" ), QStringLiteral( "road" ), QStringLiteral( "id" )};
```

Comment:
That was "descrip" before, "descript" (for shp compatibility?) now and in my case it would certainly be "description". Should this maybe check for `startsWith( tolower (candidate ) )` instead?

Reply:
It's doing a case-insensitive contains check already. I'd be happy to change "descrip" to just "desc". What do you think?

Reply:
sounds good!